parquet-column/src/main/java/org/apache/parquet/column/page/NotInPageFilteringModeException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.page;
+
+/**
+ * Exception thrown if an operation has no meaning when page filtering mode is not active.
+ * 
+ * @see PageReadStore#isInPageFilteringMode()
+ */
+public class NotInPageFilteringModeException extends IllegalStateException {
```

Comment:
I don't think this exception is well-defined. Why throw an exception when some configuration isn't set? I think what this is trying to represent is when column indexes are missing or similar cases. I'd say that the problem is trying to group all of those problems together. If an index is missing, then Parquet should fall back to normal reads. Methods that return index information should return options. And classes that can only be used with indexes should return UnsupportedOperationException or similar when they are misused.